_posts/2016-10-11-approaching-tdd-outside-android-iii.md

```diff
@@ -0,0 +1,103 @@
+
+<h2>Inner loop continuation</h2>
+To finish the <b>BankAccount</b> class, we need to implement its last public method, <b>showStatement</b>. Let's dive into the next iteration of the inner loop cycle.
+
+<ul style="display: inline-block; list-style: none; text-align: left;">
+<li><span style="color: #d32f2f; padding-right: 5px;">●</span><a href="https://github.com/CarlosMChica/AndroidBankKata/commit/0a0af116beacac964c95552dcef5ae120d2ff817">Red</a> - We created a StatementFormatter to format the statement lines. We considered the statement to be a domain concept important enough to have its own class. It acts as a first class collection around the statement lines that attracts behaviour related to the statement lines.
+The failing test ensures that when we show an account statement, the view is called with the appropriate statement lines.</li>
+
+<li><span style="color: #388e3c; padding-right: 5px;">●</span><a href="https://github.com/CarlosMChica/AndroidBankKata/commit/dd1a92aaf6b6480ac2dbfcdaf29188fdcb2b7611">Green</a> - The implementation is quite simple in this case. The BankAccount just needs to create a new statement with the transactions from the repository, pass it to the formatter to format and show the formatted statement using the view.</li>
+<li><span style="color: #1976d2; padding-right: 5px;">●</span>Refactor - Nothing to refactor here.</li>
+</ul>
+
+As a rule of thumb, once a class is done, the next step is to run the acceptance test to check the progress and to know what is the next collaborator to implement. 
+
+If we executed the acceptance test at this point, we would see that the <b>TransactionRepository</b> throws an UnsupportedOperationException. As we stated before, throwing an exception in the methods that we have not implemented yet will guide us through the feature implementation and will point us to the next collaborator to implement. It is quite important, as doing so, we have greater control over the current feature progress. We just need to follow the exceptions until the feature is fully implemented.
+
+<ul style="display: inline-block; list-style: none; text-align: left;">
+<li><span style="color: #d32f2f; padding-right: 5px;">●</span><a href="https://github.com/CarlosMChica/AndroidBankKata/commit/f1585c33ae99ed2daa3aec6a6b9b9e999e8e391e">Red</a> - TransactionRepository is a leaf node in the tree of collaborators. Meaning that, it does not collaborate with other classes. Leaf nodes can not be tested using mocks as there is no collaboration, so we need to flip to the classicist style of TDD and test through state rather than through behaviour. TransactionRepository is going to be implemented as an in-memory storage, so we decided to test it by storing a transaction and checking that the list returned in the transactions method contains the previously stored transaction.</li>
```

Comment:
Maybe capitalise and tick `Transaction` to make it easier to understand this is talking about a bank transaction rather than a technical transaction used by the database etc.